feature-libs/pickup-in-store/components/container/cart-pickup-options-container/cart-pickup-options-container.component.ts

```diff
@@ -4,44 +4,157 @@
   displayName$: Observable<string>;
+  cartId: string;
+  name: string;
+  entryNumber: number;
+  productCode: string;
+  quantity: number;
+  userId: string;
+  displayPickupLocation$: Observable<string | undefined>;
+  availableForPickup = false;
+  private displayNameIsSet = false;
+
   constructor(
     @Optional() protected outlet: OutletContextData<OrderEntry>,
-    protected storeDetails: PickupLocationsSearchFacade
+    protected activeCartFacade: ActiveCartFacade,
+    protected intendedPickupLocationService: IntendedPickupLocationFacade,
+    protected launchDialogService: LaunchDialogService,
+    protected preferredStoreService: PreferredStoreService,
+    protected pickupLocationsSearchService: PickupLocationsSearchFacade,
+    protected vcr: ViewContainerRef
   ) {
     // Intentional empty constructor
   }
 
   ngOnInit() {
     this.pickupOption$ = this.outlet?.context$.pipe(
+      filter((outletContextData) => !!outletContextData),
+      tap(
+        (outletContextData) =>
+          (this.entryNumber = outletContextData.entryNumber ?? -1)
+      ),
+      tap(
+        (outletContextData) =>
+          (this.quantity = outletContextData.quantity ?? -1)
+      ),
+      tap(
+        (outletContextData) =>
+          (this.productCode = outletContextData.product?.code || '')
+      ),
+      tap(
+        (outletContextData) =>
+          (this.name = outletContextData.deliveryPointOfService?.name ?? '')
+      ),
       map(
         (item): PickupOption =>
-          item?.deliveryPointOfService ? 'pickup' : 'delivery'
+          item.deliveryPointOfService ? 'pickup' : 'delivery'
       )
     );
 
     this.displayName$ = this.outlet?.context$.pipe(
-      map((entry) => entry?.deliveryPointOfService?.name),
+      filter((outletContextData) => !!outletContextData),
+      map((entry) => entry.deliveryPointOfService?.name),
       filter((name): name is string => !!name),
-      tap((storeName) => this.storeDetails.loadStoreDetails(storeName)),
-      switchMap((storeName) => this.storeDetails.getStoreDetails(storeName)),
-      map((store) => store?.displayName ?? '')
+      tap((storeName) =>
+        this.pickupLocationsSearchService.loadStoreDetails(storeName)
+      ),
+      switchMap((storeName) =>
+        this.pickupLocationsSearchService.getStoreDetails(storeName)
+      ),
+      map((store) => store?.displayName ?? ''),
+      tap((_displayName) => _displayName && (this.displayNameIsSet = true))
+    );
+
+    this.activeCartFacade
+      .getActive()
+      .pipe(
+        tap((cart) => (this.cartId = cart.guid ?? '')),
+        tap((cart) => (this.userId = cart.user?.uid ?? ''))
+      )
+      .subscribe();
+  }
+
+  onPickupOptionChange(pickupOption: PickupOption): void {
+    const payload: SetPickupOptionDeliveryPayload = {
+      deliveryPointOfService: {
+        name: '',
+      },
+      product: {
+        code: this.productCode,
+      },
+      quantity: this.quantity as number,
+    };
```

Comment:
The component shouldn't care about how the API request will be structured, so we shouldn't build it here. That logic should be moved down to the OCC adapter layer.